### PR TITLE
Made code clearer, commented it more, and added myself as author

### PR DESCRIPTION
**Pull request by mdew192837**
- Renamed everything in uppercase since they're essentially constants
- Added a reset function in case we need to reset anymore values
- Organized code more
- Commented more stuff

### CODE REVIEW

File: nationalsCodev1.ino [hunk 1]
```diff
@@ -1,85 +1,95 @@
...
 
+  /* Encoder */
     // Red - 5V
     // Black - GND
-    //Signal A in D2
-    const int encoder_a = 2; 
-    //Signal B in D3
-    const int encoder_b = 3; 
+    // Signal A in D2
+    const int ENCODER_A = 2;
```
Comment (mdew192837) [inline nationalsCodev1.ino:18]: Which one is this, left or right?

File: nationalsCodev1.ino [hunk 2]
```diff
@@ -1,85 +1,95 @@
...
     // Red - 5V
     // Black - GND
-    //Signal A in D2
-    const int encoder_a = 2; 
-    //Signal B in D3
-    const int encoder_b = 3; 
+    // Signal A in D2
+    const int ENCODER_A = 2; 
+    // Signal B in D3
+    const int ENCODER_B = 3;
```
Comment (mdew192837) [inline nationalsCodev1.ino:20]: Which one is this, left or right?

File: nationalsCodev1.ino [hunk 3]
```diff
@@ -1,85 +1,95 @@
...
+  // Call the reset
+  reset();
+
+  // Error handling
+  Serial.begin(9600);
 
-pinMode(pinPwm, OUTPUT); //Configures motor speed as output//
-pinMode(pinDir, OUTPUT); //Configures motor direction as output//
+  // Attach encoders
+  pinMode( ENCODER_A, INPUT_PULLUP );
```
Comment (mdew192837) [inline nationalsCodev1.ino:45]: Maybe explain why we need pullup here

File: nationalsCodev1.ino [hunk 4]
```diff
@@ -1,85 +1,95 @@
...
+  pinMode( ENCODER_B, INPUT_PULLUP );
+
+  // Attach encoder interrupt
+  attachInterrupt(0, encoderPinChangeA, CHANGE );
```
Comment (mdew192837) [inline nationalsCodev1.ino:49]: Please explain why interrupt is needed

File: nationalsCodev1.ino [hunk 5]
```diff
@@ -1,85 +1,95 @@
...
-      analogWrite(pinPwm, 50); //slow speed//
-      digitalWrite(pinDir, LOW);
-    }
+
+  // Print encoder value
+  Serial.print("Encoder value: ");
+  Serial.println(ENCODER_VALUE);
+
+  // Calculate real distance
+  REAL_DISTANCE = abs( ( ( encoder / 600 ) * ( 3.14159 * 0.123825 ) ) )
```
Comment (mdew192837) [inline nationalsCodev1.ino:67]: Can you explain this calculation?

File: nationalsCodev1.ino [hunk 6]
```diff
@@ -1,85 +1,95 @@
...
+    analogWrite(pinPwm, 50);
+    digitalWrite(pinDir, LOW);
+  }
 }
-///////////////////////////////////////////////////////////////////////////////////////////
-/// DO NOT EDIT///
-/// Used to update encoder values ///
+
+/* DO NOT EDIT - USED TO UPDATE ENCODER VALUES */
 void encoderPinChangeA() {
```
Comment (mdew192837) [inline nationalsCodev1.ino:87]: Don't really get what this function is doing. Why is it checking if they're equal?

File: nationalsCodev1.ino [hunk 7]
```diff
@@ -1,85 +1,95 @@
...
-encoder += digitalRead(encoder_a) == digitalRead(encoder_b) ? -1 : 1;
+  // Check if encoders equal to each other
+  encoder += digitalRead(ENCODER_A) == digitalRead(ENCODER_B) ? -1 : 1;
 }
 
 void encoderPinChangeB() {
```
Comment (mdew192837) [inline nationalsCodev1.ino]: Same here as above.

**Review by mdew192837 — COMMENTED** [review]
+NilaiVemula please answer my questions so I can update and amend this.

File: nationalsCodev1.ino (hunk 6 above)
Comment (NilaiVemula) [reply nationalsCodev1.ino:87]: That is how you update the encoder value.

File: nationalsCodev1.ino (hunk 5 above)
Comment (NilaiVemula) [reply nationalsCodev1.ino:67]: The encoder sends out 600 pulses per rotation. The wheels travel pi*d per rotation.

File: nationalsCodev1.ino (hunk 4 above)
Comment (NilaiVemula) [reply nationalsCodev1.ino:49]: "Good tasks for using an interrupt may include reading a rotary encoder, or monitoring user input."
https://www.arduino.cc/en/Reference/attachInterrupt

File: nationalsCodev1.ino (hunk 3 above)
Comment (NilaiVemula) [reply nationalsCodev1.ino:45]: https://www.arduino.cc/en/Tutorial/InputPullupSerial
It uses the internal pull-up resistor of the arduino. This allows it to read a signal and then output to the serial monitor.

File: nationalsCodev1.ino (hunk 2 above)
Comment (NilaiVemula) [reply nationalsCodev1.ino:20]: I need to confirm with the vehicle. It is clockwise/counter-clockwise, not left/right.

**Pull request merged**